cpp/src/parquet/properties.h

```diff
@@ -1032,6 +1035,18 @@ class PARQUET_EXPORT ArrowReaderProperties {
     }
   }
 
+  /// \brief Set the Arrow binary type to read BYTE_ARRAY columns as.
+  ///
+  /// Allowed values are Type::BINARY, Type::LARGE_BINARY and Type::BINARY_VIEW.
+  /// Default is Type::BINARY.
+  ///
+  /// If a serialized Arrow schema is found in the Parquet metadata,
+  /// this setting is ignored and the Arrow schema takes precedence
+  /// (see ArrowWriterProperties::store_schema).
+  void set_binary_type(::arrow::Type::type value) { binary_type_ = value; }
```

Comment:
So the check is delayed in `configured_binary_type` fn?

Reply:
There are many ways that users may want to direct the formation of the output schema, and unfortunately it seems difficult to provide an API that would work for all use cases. For example, what if someone wants to rely on the stored Arrow schema, but still force all binary columns to binary_view instead?

(that doesn't mean we shouldn't expose more settings, of course)